app/screens/home/search/results/file_results.tsx

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {StyleSheet, FlatList, ListRenderItemInfo, StyleProp, View, ViewStyle} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import File from '@components/files/file';
+import NoResultsWithTerm from '@components/no_results_with_term';
+import {ITEM_HEIGHT} from '@components/option_item';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {useImageAttachments} from '@hooks/files';
+import {bottomSheet, dismissBottomSheet} from '@screens/navigation';
+import NavigationStore from '@store/navigation_store';
+import {isImage, isVideo} from '@utils/file';
+import {fileToGalleryItem, openGalleryAtIndex} from '@utils/gallery';
+import {bottomSheetSnapPoint} from '@utils/helpers';
+import {getViewPortWidth} from '@utils/images';
+import {TabTypes} from '@utils/search';
+import {preventDoubleTap} from '@utils/tap';
+
+import FileOptions from './file_options';
+import {HEADER_HEIGHT} from './file_options/header';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        marginHorizontal: 20,
+    },
+});
+
+type Props = {
+    canDownloadFiles: boolean;
+    fileChannels: ChannelModel[];
+    fileInfos: FileInfo[];
+    publicLinkEnabled: boolean;
+    scrollPaddingTop: number;
+    searchValue: string;
+}
+
+const galleryIdentifier = 'search-files-location';
+
+const FileResults = ({
+    canDownloadFiles,
+    fileChannels,
+    fileInfos,
+    publicLinkEnabled,
+    scrollPaddingTop,
+    searchValue,
+}: Props) => {
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const insets = useSafeAreaInsets();
+    const [lastViewedIndex, setLastViewedIndex] = useState<number | undefined>(undefined);
+
+    const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, flexGrow: 1}), [scrollPaddingTop]);
+    const {images: imageAttachments, nonImages: nonImageAttachments} = useImageAttachments(fileInfos, publicLinkEnabled);
+    const channelNames = useMemo(() => fileChannels.reduce<{[id: string]: string | undefined}>((acc, v) => {
+        acc[v.id] = v.displayName;
+        return acc;
+    }, {}), [fileChannels]);
+
+    const containerStyle = useMemo(() => {
+        return {top: fileInfos.length ? 8 : 0};
+    }, [fileInfos]);
+
+    const filesForGallery = useMemo(() => imageAttachments.concat(nonImageAttachments),
+        [imageAttachments, nonImageAttachments]);
+
+    const orderedFilesForGallery = useMemo(() => (
+        filesForGallery.sort((a: FileInfo, b: FileInfo) => {
+            return (b.create_at || 0) - (a.create_at || 0);
+        })
+    ), [filesForGallery]);
```

Comment:
Just a thought here and no action is required - what is really being memoized? What is the cost of not using so many useMemo ?